src/Avalonia.Controls/SelectionModel.cs

```diff
@@ -452,9 +489,15 @@ private void RaisePropertyChanged(string propertyName)
             PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
         }
 
-        public void OnSelectionInvalidatedDueToCollectionChange()
+        public void OnSelectionInvalidatedDueToCollectionChange(
+            IEnumerable<object>? removedItems)
         {
-            OnSelectionChanged();
+            var e = new SelectionModelSelectionChangedEventArgs(
```

Comment:
>new SelectionModelSelectionChangedEventArg [](start = 20, length = 42)

This was cached to avoid repeated allocations. Not sure if that is a concern in this project.

Reply:
Yeah I removed the caching because caching could cause a problem if a `SelectionChanged` event handler made a change to the selection itself. However thinking about this some more, because the added/removed items are lazily-evaluated, this could still break...